src/coalescenceml/artifact_store/local_artifact_store.py

```diff
@@ -0,0 +1,137 @@
+
+
+class LocalArtifactStore(BaseArtifactStore):
+    """Artifact Store for local artifacts."""
+
+    # Class Configuration
+    FLAVOR: ClassVar[str] = "local"
+    SUPPORTED_SCHEMES: ClassVar[Set[str]] = {""}
+
+    @staticmethod
+    def open(name: PathType, mode: str = "r") -> Any:
+        """Open a file at the given path."""
+        return open(name, mode=mode)
+
+    @staticmethod
+    def copyfile(src: PathType, dst: PathType, overwrite: bool = False) -> None:
+        """Copy a file from the source to the destination."""
+        if not overwrite and os.path.exists(dst):
+            raise FileExistsError(
+                f"Destination file {str(dst)} already exists and argument "
+                f"`overwrite` is false."
+            )
+        shutil.copyfile(src, dst)
+
+    @staticmethod
+    def exists(path: PathType) -> bool:
+        """Returns `True` if the given path exists."""
+        return os.path.exists(path)
+
+    @staticmethod
+    def glob(pattern: PathType) -> List[PathType]:
+        """Return the paths that match a glob pattern."""
+        return glob.glob(pattern)
+
+    @staticmethod
+    def isdir(path: PathType) -> bool:
+        """Returns whether the given path points to a directory."""
+        return os.path.isdir(path)
+
+    @staticmethod
+    def listdir(path: PathType) -> List[PathType]:
+        """Returns a list of files under a given directory in the filesystem."""
+        return os.listdir(path)
+
+    @staticmethod
+    def makedirs(path: PathType) -> None:
+        """Make a directory at the given path, recursively creating parents."""
+        os.makedirs(path, exist_ok=True)
+
+    @staticmethod
+    def mkdir(path: PathType) -> None:
+        """Make a directory at the given path; parent directory must exist."""
+        os.mkdir(path)
+
+    @staticmethod
+    def remove(path: PathType) -> None:
+        """Remove the file at the given path. Dangerous operation."""
+        os.remove(path)
+
+    @staticmethod
+    def rename(src: PathType, dst: PathType, overwrite: bool = False) -> None:
+        """Rename source file to destination file.
+        Args:
+            src: The path of the file to rename.
+            dst: The path to rename the source file to.
+            overwrite: If a file already exists at the destination, this
+                method will overwrite it if overwrite=`True`
+        """
+        if not overwrite and os.path.exists(dst):
+            raise FileExistsError(
+                f"Destination path {str(dst)} already exists and argument "
+                f"`overwrite` is false."
+            )
+        os.rename(src, dst)
+
+    @staticmethod
+    def rmtree(path: PathType) -> None:
+        """Deletes dir recursively. Dangerous operation."""
+        shutil.rmtree(path)
+
+    @staticmethod
+    def stat(path: PathType) -> Any:
+        """Return the stat descriptor for a given file path."""
+        return os.stat(path)
+
+    @staticmethod
```

Comment:
Maybe mention what options do? Dunno if you did doc this elsewhere I'm just going down the files